packages/paypal-express-integration/src/paypal-express-button-strategy.ts

```diff
@@ -0,0 +1,174 @@
+            options.methodId,
+        ));
+
+        if (!paypalOptions) {
+            throw new InvalidArgumentError();
+        }
+
+        if (!paymentMethod) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+       if (!paymentMethod || !paymentMethod.config.merchantId) {
+           throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+       }
+
+        const paypal = await this.paypalExpressIntegrationService.loadPaypal(paymentMethod.config.merchantId);
+
+        const merchantId = paymentMethod.config.merchantId;
+        const env = paymentMethod.config.testMode ? 'sandbox' : 'production';
+        const clientToken: PaypalClientToken = { [env]: paypalOptions.clientId };
+
+        const fundingCreditOption = paypal.FUNDING.CREDIT || 'credit';
+        const allowedSources = paypalOptions.allowCredit ? [fundingCreditOption] : [];
+        const disallowedSources = !paypalOptions.allowCredit ? [fundingCreditOption] : [];
+
+        return paypal.Button.render(
+            {
+                env,
+                client: clientToken,
+                commit: paypalOptions.shouldProcessPayment,
+                funding: {
+                    allowed: allowedSources,
+                    disallowed: disallowedSources,
+                },
+                style: {
+                    shape: PaypalButtonStyleShapeOption.RECT,
+                    ...pick(
+                        paypalOptions.style,
+                        'layout',
+                        'color',
+                        'label',
+                        'shape',
+                        'tagline',
+                        'fundingicons',
+                    ),
+                    size:
+                        paymentMethod.id === 'paypalexpress' &&
+                        paypalOptions.style?.size === 'small'
+                            ? PaypalButtonStyleSizeOption.RESPONSIVE
+                            : paypalOptions.style?.size,
+                },
+                payment: (_, actions) =>
+                    this._setupPayment(merchantId, actions, paypalOptions.onPaymentError),
+                onAuthorize: (data, actions) =>
+                    this._tokenizePayment(
+                        data,
+                        actions,
+                        paypalOptions.shouldProcessPayment,
+                        paypalOptions.onAuthorizeError,
+                    ),
+            },
+            options.containerId,
+        );
+    }
+
+    deinitialize(): Promise<void> {
+        this._paymentMethod = undefined;
+
+        return Promise.resolve();
+    }
+    private async  _setupPayment(
+        merchantId: string,
+        actions?: PaypalActions,
+        onError?: (error: StandardError) => void,
+    ): Promise<string> {
+        if (!actions) {
+            throw new NotInitializedError(NotInitializedErrorType.CheckoutButtonNotInitialized);
+        }
```

Comment:
Add an extra empty line below, please

Reply:
Do you mean above? It is already added empty line below

Reply:
Please, double check this message